hardware_interface/src/resource_manager.cpp

```diff
@@ -25,16 +25,51 @@
 #include "hardware_interface/actuator.hpp"
 #include "hardware_interface/actuator_interface.hpp"
 #include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_component_info.hpp"
 #include "hardware_interface/sensor.hpp"
 #include "hardware_interface/sensor_interface.hpp"
 #include "hardware_interface/system.hpp"
 #include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
 #include "lifecycle_msgs/msg/state.hpp"
 #include "pluginlib/class_loader.hpp"
 #include "rcutils/logging_macros.h"
 
+#define COMPONENT_NAME_COMPARE [&](const auto & name) { return name == component.get_name(); }
+#define COMPONENT_NAME_COMPARE_INV \
+  [&](const auto & component) { return component.get_name() == component_name; }
+
+#define MOVEMENT_INTERFACES_COMPARE                                                         \
```

Comment:
why a define on these and not a simple `using` or just assigning the lambda toa const var?

Reply:
This is because the lambda is outside of class and I would like to have access to the current context "[&]". Will check if there is something better to do here...